packages/eas-cli/src/credentials/ios/appstore/bundleIdCapabilities.ts

```diff
@@ -0,0 +1,415 @@
+import {
+  BundleId,
+  CapabilityOptionMap,
+  CapabilityType,
+  CapabilityTypeDataProtectionOption,
+  CapabilityTypeOption,
+} from '@expo/apple-utils';
+import { JSONObject, JSONValue } from '@expo/json-file';
+
+import Log from '../../../log';
+
+type GetOptionsMethod<T extends CapabilityType = any> = (
+  entitlement: JSONValue,
+  entitlementsJson: JSONObject
+) => CapabilityOptionMap[T];
+
+const validateBooleanOptions = (options: any): boolean => {
+  return typeof options === 'boolean';
+};
+
+const validateStringArrayOptions = (options: any): boolean => {
+  return Array.isArray(options) && options.every(option => typeof option === 'string');
+};
+
+const createValidateStringOptions = (allowed: string[]) => (options: any): boolean => {
+  return allowed.includes(options);
+};
+
+const createValidateStringArrayOptions = (allowed: string[]) => (options: any): boolean => {
+  return Array.isArray(options) && options.every(option => allowed.includes(option));
+};
+
+const validateDevProdString = createValidateStringOptions(['development', 'production']);
+
+const getBooleanOptions: GetOptionsMethod = entitlement => {
+  return entitlement === true ? CapabilityTypeOption.ON : CapabilityTypeOption.OFF;
+};
+
+const getDefinedOptions: GetOptionsMethod = entitlement => {
+  return entitlement ? CapabilityTypeOption.ON : CapabilityTypeOption.OFF;
+};
+
+export async function syncCapabilitiesForEntitlementsAsync(
```

Comment:
Can you please break this function down to a few smaller? It's really difficult to follow it.